app/main.py

```diff
@@ -97,37 +99,70 @@ def grab_secrets(self):
 
         if secrets_keys is not None:
             for key_info in filter(None, secrets_keys.split(';')):
-                key_name, _, key_version = key_info.strip().partition(':')
-                _logger.info('Retrieving secret name:%s with version: %s', key_name, key_version)
+                key_name, key_version, cert_filename, key_filename = self._split_keyinfo(key_info)
+                _logger.info('Retrieving cert name:%s with version: %s output certFileName: %s keyFileName: %s', key_name, key_version, cert_filename, key_filename)
                 secret = client.get_secret(vault_base_url, key_name, key_version)
                 output_path = os.path.join(self._secrets_output_folder, key_name)
```

Comment:
shouldn't this support key_filename?

Reply:
Do you mean a filename for secret?

Reply:
I think this is cert_filename (if the user gave a name to use)

Reply:
@raghu017 - yes, if we support file names for secrets that aren't backing a cert.